shared/src/business/entities/cases/Case.ts

```diff
@@ -1921,6 +1937,22 @@ export class Case extends JoiValidationEntity {
     return this;
   }
 
+  /**
+   * sets the remote trial granted date and boolean flag
+   * @param {string | null | undefined} remoteTrialGrantedDate the date when motion to proceed remotely was granted, or null/undefined to clear
+   * @returns {Case} this case entity
+   */
+  setRemoteTrialGrantedDate(remoteTrialGrantedDate?: string | null) {
+    const hasDate = Boolean(
+      remoteTrialGrantedDate &&
+        typeof remoteTrialGrantedDate === 'string' &&
+        remoteTrialGrantedDate.trim() !== '',
+    );
+    this.remoteTrialGranted = hasDate;
+    this.remoteTrialGrantedDate = hasDate ? remoteTrialGrantedDate : null;
+    return this;
+  }
```

Comment:
Since we have a known return type, maybe have an explicit declaration instead of letting typescript infer it? IMO it's fine either way but I like explicit types in the method signature.

```suggestion
  setRemoteTrialGrantedDate(remoteTrialGrantedDate?: string | null): Case {
    const hasDate = Boolean(
      remoteTrialGrantedDate &&
        typeof remoteTrialGrantedDate === 'string' &&
        remoteTrialGrantedDate.trim() !== '',
    );
    this.remoteTrialGranted = hasDate;
    this.remoteTrialGrantedDate = hasDate ? remoteTrialGrantedDate : null;
    return this;
  }
```

Reply:
Thank you @zacharybrown204 I completely agree. I all for readability and and explicit declarations. We will return the type for the function.